BlockServer/alarm/load_alarm_config.py

```diff
@@ -0,0 +1,58 @@
+import subprocess
+import threading
+import os
+import time
+
+
+class AlarmConfigLoader(object):
+
+    # Instance of this singleton
+    instance = None
+    # The subprocess that the alarm server reloads itself in
+    process = None
+    # Number of seconds to delay the reload by
+    # (the blockserver must have finished loading the configuration before the alarm server can be restarted)
+    delay = 20
+
+    @staticmethod
+    def get_instance():
+
+        if AlarmConfigLoader.instance is None:
+            AlarmConfigLoader.instance = AlarmConfigLoader()
+
+        return AlarmConfigLoader.instance
+
+    def _load(self):
+        """
+        Simple wrapper around a batch script which reloads the configuration in the alarm server.
+        """
+
+        # Terminate the last batch job if it is still running
+        # (prevents two instances from running concurrently which might cause problems).
+        if self.process is not None:
+            self.process.kill()
+
+        print "Alarm server will update in " + self.delay + " seconds\n"
+        time.sleep(self.delay)
+        print "Alarm server updating...\n"
+
+        filepath = os.path.join('C:\\', 'Instrument', 'Apps', 'EPICS', 'CSS', 'master', 'AlarmServer', 'run_alarm_server.bat')
+
+        try:
+            with open(filepath, 'r') and open(os.devnull) as devnull:
+                self.process = subprocess.Popen(filepath, stderr=devnull)
+                self.process.communicate(input=None)
```

Comment:
Why have you used communicate here rather than call here? If it's because of the warning about deadlocks as documented here:

https://docs.python.org/2/library/subprocess.html

please could you include a comment so that it doesn't get changed in the future